libs/report/ingestion/src/lib/ingestion-manifest.spec.ts

```diff
@@ -0,0 +1,81 @@
+import { existsSync, readFileSync, statSync } from 'node:fs';
+import { promptToProceedUploadFile } from './prompts';
+import { findManifestFile } from './send-manifest';
+
+jest.mock('node:fs', () => ({
+  readFileSync: jest.fn(),
+  statSync: jest.fn(),
+  existsSync: jest.fn(),
+}));
+
+jest.mock('./prompts', () => ({
+  promptToProceedUploadFile: jest.fn(),
+}));
+
+describe('Telemetry Functions', () => {
+  it('should find manifest files successfully', async () => {
+    const mockFileName = 'package.json';
+    const mockFileData = '{"name": "test package"}';
+    const mockFileStat = { size: 1024 };
+    (existsSync as jest.Mock).mockReturnValue(true);
+    (statSync as jest.Mock).mockReturnValue(mockFileStat);
+    (readFileSync as jest.Mock).mockReturnValue(mockFileData);
+    (promptToProceedUploadFile as jest.Mock).mockResolvedValue(true);
+
+    const result = await findManifestFile();
+
+    expect(result).toEqual({ name: mockFileName, data: mockFileData });
+    expect(promptToProceedUploadFile).toHaveBeenCalledWith(mockFileName);
+  });
+
+  it('should warn if manifest files do not exist', async () => {
+    (existsSync as jest.Mock).mockReturnValue(false);
+    (promptToProceedUploadFile as jest.Mock).mockResolvedValue(true);
+    const consoleWarnSpy = jest.spyOn(console, 'log').mockImplementation();
```

Comment:
BTW, the variable name makes it sound like this is `console.warn()` (which it's not) :grin:

Reply:
It used to be 😆 imma rename it